api/capi/include/nntrainer.h

```diff
@@ -227,6 +290,64 @@ int ml_nnoptimizer_delete(ml_nnopt_h opt);
  */
 int ml_nnoptimizer_set_property(ml_nnopt_h opt, ...);
 
+/**
+ * @brief Create a dataset with generators to feed to a neural network.
+ * @details Use this function to create a Neural Network Dataset using
+ * generators. The generators will provide data of the given bytes representing
+ * a single input.
+ * @since_tizen 6.x
+ * @param[out] dataset The NNTrainer Dataset handler from the given description.
+ * @param[in] train_cb The dataset generator for training.
+ * @param[in] valid_cb The dataset generator for validating. Can be null.
+ * @param[in] test_cb The dataset generator for testing. Can be null.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid Parameter.
+ */
+int ml_nndataset_create_with_generator(ml_nndataset_h *dataset,
+    dataset_generator train_cb, dataset_generator valid_cb,
+    dataset_generator test_cb);
+
+/**
+ * @brief Create a dataset with files to feed to a neural network.
+ * @details Use this function to create a Neural Network Dataset using
+ * files.
+ * @since_tizen 6.x
+ * @param[out] dataset The NNTrainer Dataset handler from the given description.
+ * @param[in] train_cb The dataset file for training.
+ * @param[in] valid_cb The dataset file for validating. Can be null.
+ * @param[in] test_cb The dataset file for testing. Can be null.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid Parameter.
+ */
+int ml_nndataset_create_with_file(ml_nndataset_h *dataset,
+    const char * train_file, const char * valid_file, const char * test_file);
+
+/**
+ * @brief Delete the neural network dataset.
+ * @details Use this function to delete dataset. Fails if dataset is owned by a
+ * model.
+ * @since_tizen 6.x
+ * @param[in] dataset The dataset handler.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid Parameter.
+ */
+int ml_nndataset_delete(ml_nndataset_h dataset);
+
+/**
+ * @brief Set the neural network dataset Property.
+ * @details Use this function to set dataset Property.
```

Comment:
what kind of data set properties are? is it something like buffer_size, mini_batch size, etc.?

Reply:
Yes - buffer size, num data loader threads, shuffle, etc. 
This function allows to extend the properties later if needed.